sheetshuttle/util.py

```diff
@@ -26,6 +26,19 @@ def gh_token_exists() -> bool:
     return False
 
 
+def extract_sheet_id(url: str) -> str:
+    """Extract a sheet ID from a Google Sheets URL."""
+    try:
+        url_segments = url.split("/")
+        # Return the url_segment following the segment that contains only "d"
+        idx = url_segments.index("d")
+        return url_segments[idx + 1]
+    except IndexError as exc:
+        raise Exception(
+            f"The URL {url} is not valid for Google Sheets ID extraction."
+        ) from exc
```

Comment:
I think this error handling is a good start. There might be few things that we didn't handle that will come up during testing but this is good for now.